src/test/stateful_modelinstance_test.cpp

```diff
@@ -315,15 +391,20 @@ TEST_F(StatefulModelInstance, PreprocessingIntermediateRequest) {
         // Inject sequence with newState as the last state written to sequence memory state
         ovms::model_memory_state_t memoryState;
         addState(memoryState, "state", shape, newState);
-        modelInstance.injectSequence(sequenceId, memoryState);
+        modelInstance->injectSequence(sequenceId, memoryState);
 
-        // Perform preprocessing (load state from sequence to infer request)
-        modelInstance.preInferenceProcessing(inferRequest, sequenceProcessingSpec);
+        tensorflow::serving::PredictResponse response;
+
+        modelInstance->postInferenceProcessing(&response, inferRequest, sequenceProcessingSpec);
+
+        auto& output = (*response.mutable_outputs())["sequence_id"];
+        EXPECT_EQ(output.uint64_val_size(), 1);
+        EXPECT_EQ(output.uint64_val(0), sequenceId);
 
         // Check if InferRequest memory state has been updated to sequence memory state
```

Comment:
In case of postprocessing InferRequest memory state does not get updated, sequence memory state does. Here we should check if sequence contains the same memory state as InferRequest.